sleap/nn/data/instance_cropping.py

```diff
@@ -42,12 +42,23 @@ def find_instance_crop_size(
     # Calculate crop size
     min_crop_size_no_pad = min_crop_size - padding
     max_length = 0.0
-    for inst in labels.user_instances:
-        pts = inst.points_array
-        pts *= input_scaling
-        max_length = np.maximum(max_length, np.nanmax(pts[:, 0]) - np.nanmin(pts[:, 0]))
-        max_length = np.maximum(max_length, np.nanmax(pts[:, 1]) - np.nanmin(pts[:, 1]))
-        max_length = np.maximum(max_length, min_crop_size_no_pad)
+    for lf in labels:
+        for inst in lf.user_instances:
+            pts = inst.points_array
+
+            pts[pts < 0] = np.NaN
+            height, width = lf.image.shape[:2]
+            pts[:, 0][pts[:, 0] > width - 1] = np.NaN
+            pts[:, 1][pts[:, 1] > height - 1] = np.NaN
+
+            pts *= input_scaling
+            max_length = np.maximum(
+                max_length, np.nanmax(pts[:, 0]) - np.nanmin(pts[:, 0])
+            )
```

Comment:
This leaves a possibility (albeit unlikely) for a `max_length` of `np.nan` if we have an all-nan instance for some or other reason (possibly because of the rumours spread [here](https://github.com/talmolab/sleap/issues/640#issuecomment-1419990083)) - although, on the same point, we currently just filter nan instances. 

Even so, suggestion:
```suggestion
            max_length: float = np.nanmax(
                [max_length, np.nanmax(pts[:, 0]) - np.nanmin(pts[:, 0])]
            )
```